pkg/nsx/services/vpc/clean_avi.go

```diff
@@ -0,0 +1,65 @@
+package vpc
+
+import (
+	"context"
+	"errors"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx"
+	nsxutil "github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
+)
+
+type (
+	mapInterface = map[string]interface{}
+)
+
+const (
+	PolicyAPI                = "policy/api/v1"
+	AviSubnetPortsPathSuffix = "/subnets/_AVI_SUBNET--LB/ports/"
+)
+
+func appendIfNotExist(slice []string, s string) []string {
+	for _, item := range slice {
+		if item == s {
+			return slice
+		}
+	}
+	return append(slice, s)
+}
+
+func httpGetAviResources(cluster *nsx.Cluster, vpcPath string) ([]string, error) {
```

Comment:
How about use a sets.NewString() instead of []string to avoid calling appendIfNotExist?
BTW, will the path has the same value?

Reply:
ok, vpcPath different value.